### PR TITLE
kong2kic changes after code review

### PR DESCRIPTION
**Pull request by battlebyte**
This change incorporates the suggestions from 
[@Gabriele Gerbino](https://kongstrong.slack.com/team/U02SAMXFYUE)
 
[@travenichka](https://kongstrong.slack.com/team/U8T62CPL0)
 
[@michal](https://kongstrong.slack.com/team/U0166UFHQKS)
 
[@Greg](https://kongstrong.slack.com/team/U041AC47PNV)

- Use of KongUpstreamPolicy in 3.x
- Use labels for credential type in 3.x
- No status or metadata.creationTimestamp fields are in the output

The new command interface is:

```
Convert Kong configuration files to Kong Ingress Controller (KIC) manifests.

The kong2kic subcommand transforms Kong’s configuration files, written in the deck format,
into Kubernetes manifests suitable for the Kong Ingress Controller. By default kong2kic generates
manifests for KIC v3.x using the Kubernetes Gateway API. Only HTTP/HTTPS routes are supported.

Usage:
  deck file kong2kic [flags]

Flags:
      --class-name string    Value to use for "kubernetes.io/ingress.class" ObjectMeta.Annotations and for
                             		"parentRefs.name" in the case of HTTPRoute. (default "kong")
  -f, --format string        output file format: json or yaml. (default "yaml")
  -h, --help                 help for kong2kic
      --ingress              Use Kubernetes Ingress API manifests instead of Gateway API manifests.
      --kicv2                Generate manifests compatible with KIC v2.x.
  -o, --output-file string   Output file to write. Use - to write to stdout. (default "-")
  -s, --state string         decK file to process. Use - to read from stdin. (default "-")
```

### CODE REVIEW

**Comment by mheap** [comment]
Aha! I'm not on @Kong/team-deck. Could someone please merge? This is targeting the `kic` branch, not `main` and we'll do a final review on #1050

**Comment by battlebyte** [comment]
My battlebyt/kic fork is already in sync with Kong:kic branch, correct?

**Comment by czeslavo** [comment]
My suggestion regarding dropping `status` and `creationTimestamp` fields has been addressed. 👍 I'm not sure about failing tests though - as I can see they're passing on https://github.com/Kong/deck/pull/1050 so probably there's something to be fixed here before merging to `kic` branch.

**Review by czeslavo — APPROVED** [review]

**Comment by Tieske** [comment]
> My battlebyt/kic fork is already in sync with Kong:kic branch, correct?

Doesn't look like it. If you look at the commits in this PR, they have a lot of work that is also in the base branch.

**Comment by battlebyte** [comment]
Sorry @Tieske , you're right. This pull request now has 1 squashed commit that continues after the last commit (f419701) in PR https://github.com/Kong/deck/pull/1050.

**Comment by Tieske** [comment]
@battlebyte I fixed the linter issues, and then accidentally merged this PR because its name is the same as the base branch.

Please ensure you change the branch name next time. (if you have no other pending work, my suggestion would be to delete your clone and create a new one all together)

File: kong2kic/types.go [hunk 1]
```diff
@@ -229,3 +228,31 @@ func (k KICContent) marshalKICContentToJSON() ([]byte, error) {
...
+	if err != nil {
+		return nil, err
+	}
+	genericObj := map[string]interface{}{}
+	if err := json.Unmarshal(objBytes, &genericObj); err != nil {
+		return nil, err
+	}
+
+	delete(genericObj, "status")
+	delete(genericObj["metadata"].(map[string]interface{}), "creationTimestamp")
```
[the remark this answers is not in the document]
Comment (Tieske) [reply kong2kic/types.go:244]: fixed this comment after merging it in the base barnch

**Pull request merged**